### ISSUE TITLE
Zero area polygons

### ISSUE THREAD

**Issue by erikwijmans** [recastnavigation/recastnavigation#414]
Question:  Are zero area polygons allowed/expected?

Context:  I have a navigation mesh that has a couple of zero area polygons.  These polygons are marked as navigable and thus can be chosen by moveAlongSurface.  Other methods in detour then fail with the zero area polygons (i.e. getPolyHeight).  Manually marking them as not navigable fixes my issues, fortunately.

If these are not expected, I can get you the navmesh and source 3D mesh.

**Comment by memononen** [comment]
Zero area polygons should not happen. A screenshot of the case would be a
good starting point to see where things might go wrong.

--mikko

**Comment by erikwijmans** [comment]
Screen shots of the mesh with a wireframe rendering of the underlying triangles:
<img width="803" alt="screen_shot_2" src="https://user-images.githubusercontent.com/12722529/70340960-d718c900-181f-11ea-8f0b-4106c02826a4.png">
<img width="799" alt="screen_shot_1" src="https://user-images.githubusercontent.com/12722529/70340963-d97b2300-181f-11ea-8aca-803565482219.png">

It is a little hard to tell where the exact issue is, but I believe it is the odd piece of geometry in the black box.

**Comment by memononen** [comment]
I'm not sure what I'm looking at. Tha does not look like the navmesh in
those pictures.

**Comment by erikwijmans** [comment]
Yeah, it's the source mesh itself.  I am not sure how to visualize the navmesh.  I couldn't figure out how to point the recast demo viewer to this mesh so that I could compile and visual the navmesh.

**Comment by erikwijmans** [comment]
We recently added navmesh visualization to our codebase now I can actually get you a screenshot of the navmesh :)

<img width="797" alt="Screen Shot 2020-06-14 at 5 52 49 PM" src="https://user-images.githubusercontent.com/12722529/84605060-5c80d400-ae68-11ea-8a11-d063b698381b.png">


This is showing the wireframe of the triangles from the detail polymesh -- extracted similarly to the code here: https://github.com/recastnavigation/recastnavigation/blob/df27e4eb1a4ade9912f8b7d75c25769a3193dbd0/Detour/Source/DetourNavMesh.cpp#L701-L709

That central point in the black box is where the zero area polygon is (obviously it doesn't render quite right).

**Comment by grahamboree** [comment]
Hi @erikwijmans!  Is there any possibility of getting some kind of source geometry or repro case for this issue?  Recast should never generate zero-area polygons so we're keen to figure out what the issue is and how to prevent it.
